core/src/main/kotlin/entity/Asset.kt

```diff
@@ -0,0 +1,90 @@
+package dev.kord.core.entity
+
+import dev.kord.common.entity.Snowflake
+import dev.kord.core.Kord
+import dev.kord.core.KordObject
+import dev.kord.rest.Image
+import dev.kord.rest.route.CdnUrl
+import dev.kord.rest.route.DiscordCdn
+
+public sealed class Asset(
+    public val format: Image.Format,
+    public val cdnUrl: CdnUrl,
+    override val kord: Kord
+) : KordObject {
+    public val url: String
+        get() = cdnUrl.toUrl {
+            this.format = this@Asset.format
+        }
+
+    public val animated: Boolean get() = format is Image.Format.GIF
+
+    public suspend fun getImage(): Image = Image.fromUrl(kord.resources.httpClient, cdnUrl.toUrl())
```

Comment:
```suggestion
    public suspend fun getImage(): Image = Image.fromUrl(kord.resources.httpClient, url)
```
this should use `Asset.format`, otherwise the default `Image.Fornat.WEBP` will be used (see `CdnUrl.toUrl()`